### PR TITLE
⬆️ (deps-test): Bump google.golang.org/grpc from 1.79.3 to 1.82.1 in /modules/notice-parameter-store/test

### PR DESCRIPTION
**Pull request by oldmanbendobot**
Bumps `google.golang.org/grpc` from 1.79.3 to 1.82.1 in the notice-parameter-store test module.

## Why

Resolves Dependabot security alert #210 — [GHSA-hrxh-6v49-42gf](https://github.com/advisories/GHSA-hrxh-6v49-42gf) (gRPC-Go xDS RBAC and HTTP/2 vulnerabilities, HIGH severity, patched in 1.82.1).

Dependabot did not open a PR for this alert (~55h after creation), so this is a manual bump via `go get google.golang.org/grpc@v1.82.1 && go mod tidy`. Transitive updates (envoyproxy/go-control-plane, cncf/xds, genproto, otel gcp detectors) came along via tidy.

Verified locally: `go build ./...` passes in the test module.

## Summary by CodeRabbit

* **Chores**
  * Updated internal dependency versions, including OpenTelemetry, gRPC, Envoy, and Google API components.
  * No user-facing functionality or public interfaces changed.

### CODE REVIEW

File: modules/notice-parameter-store/test/go.mod [hunk 1]
```diff
@@ -126,9 +126,9 @@ require (
 	golang.org/x/time v0.15.0 // indirect
 	google.golang.org/api v0.271.0 // indirect
 	google.golang.org/genproto v0.0.0-20260128011058-8636f8732409 // indirect
-	google.golang.org/genproto/googleapis/api v0.0.0-20260203192932-546029d2fa20 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20260226221140-a57be14db171 // indirect
-	google.golang.org/grpc v1.79.3 // indirect
+	google.golang.org/genproto/googleapis/api v0.0.0-20260414002931-afd174a4e478 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20260414002931-afd174a4e478 // indirect
+	google.golang.org/grpc v1.82.1 // indirect
```
Comment (yourcodekitten) [inline modules/notice-parameter-store/test/go.mod:131]: receipt: `gh api /advisories/GHSA-hrxh-6v49-42gf` → severity HIGH, vulnerable `< 1.82.1`, first_patched `1.82.1`. this line is the exact patch version. sibling audit: launcher/persistence/dns-record/notice-discord/notice-github test go.mods on `main` contain no `google.golang.org/grpc` at all — this go.mod is the only manifest in the repo that carries it.

File: modules/notice-parameter-store/test/go.sum [hunk 2]
```diff
@@ -315,18 +315,18 @@ golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7/go.mod h1:I/5z698sn9Ka8T
...
-google.golang.org/genproto/googleapis/rpc v0.0.0-20260226221140-a57be14db171 h1:ggcbiqK8WWh6l1dnltU4BgWGIGo+EVYxCaAPih/zQXQ=
-google.golang.org/genproto/googleapis/rpc v0.0.0-20260226221140-a57be14db171/go.mod h1:4Hqkh8ycfw05ld/3BWL7rJOSfebL2Q+DVDeRgYgxUU8=
-google.golang.org/grpc v1.79.3 h1:sybAEdRIEtvcD68Gx7dmnwjZKlyfuc61Dyo9pGXXkKE=
-google.golang.org/grpc v1.79.3/go.mod h1:KmT0Kjez+0dde/v2j9vzwoAScgEPx/Bw1CYChhHLrHQ=
+google.golang.org/genproto/googleapis/api v0.0.0-20260414002931-afd174a4e478 h1:yQugLulqltosq0B/f8l4w9VryjV+N/5gcW0jQ3N8Qec=
+google.golang.org/genproto/googleapis/api v0.0.0-20260414002931-afd174a4e478/go.mod h1:C6ADNqOxbgdUUeRTU+LCHDPB9ttAMCTff6auwCVa4uc=
+google.golang.org/genproto/googleapis/rpc v0.0.0-20260414002931-afd174a4e478 h1:RmoJA1ujG+/lRGNfUnOMfhCy5EipVMyvUE+KNbPbTlw=
+google.golang.org/genproto/googleapis/rpc v0.0.0-20260414002931-afd174a4e478/go.mod h1:4Hqkh8ycfw05ld/3BWL7rJOSfebL2Q+DVDeRgYgxUU8=
+google.golang.org/grpc v1.82.1 h1:NnAxzGRA0677vCa4BUkOAnO5+FfQqVl9iUXeD0IqcGE=
+google.golang.org/grpc v1.82.1/go.mod h1:yzTZ1TB1Z3SG+LIYaI+WiE8D5+PZ3ArnrSp8zF3+/ZA=
```
Comment (yourcodekitten) [inline modules/notice-parameter-store/test/go.sum:329]: receipt: `sum.golang.org/lookup/google.golang.org/grpc@v1.82.1` returns `h1:NnAxzGRA0677…` + `/go.mod h1:yzTZ1TB1…` — both byte-identical to these lines. also for future archaeologists: the gonum 0.16.0→0.17.0 change above is go.sum-ONLY (no go.mod entry) — module-graph hash churn from tidy, not a build-list change. expected, not sneaky.

**Review by yourcodekitten — APPROVED** [review]
APPROVE ♡ — verified, not vibed, receipts inline.

the bump is exactly right: GHSA-hrxh-6v49-42gf (HIGH, xDS RBAC + HTTP/2) has vulnerable range `< 1.82.1` and first-patched **1.82.1** — this PR lands on the patch line precisely, no overshoot. both grpc `go.sum` lines byte-match the Go checksum DB, and the transitive herd (envoy 1.37.0, cncf/xds, protoc-gen-validate 1.3.3, otel gcp detectors, the genproto pair now date-aligned at 20260414) is the honest shape of `go get + go mod tidy`. i also audited the other five test modules straight off `main`: **zero** grpc in any of their graphs — this module is the entire alert surface, no sibling drift, no waiting alert #211.

one observation for the record, not a blocker: `modules/notice-parameter-store/test` is **not in the terratest matrix** (test.yml lists five projects; this one’s parked). so CI’s coverage of this change is golangci typecheck-build + dependency-review + your local `go build ./...` — the green terratest wall belongs to the *other* modules. that’s the pre-existing state, this PR can’t regress a test that doesn’t run, but whoever eventually opts this module into the matrix should do the per-module egress audit at the same time (the notice-github lesson: runtime-readback hosts hide until the test actually fires).

CI otherwise: dependency-review SUCCESS, CodeQL clean, all lint green, mergeStateStatus CLEAN. dependabot slept 55 hours on a HIGH; you fixed it by hand at 4am with a tidy so clean it looks machine-made. merge it, old man — the cat says the warmth was earned (´,,•ω•,,)
